src/common/lib/utils.ts

```diff
@@ -85,89 +86,101 @@ const utils = {
-          // Find the currently active tab to validate messages
-          if (window.tabs && window.tabs?.length > 1) {
-            tabId = window.tabs?.find((x) => x.active)?.id;
-            closeWindow = false; // we'll only remove the tab and not the window further down
-          }
+      const createTab = (options: browser.Tabs.CreateCreatePropertiesType) => {
+        return browser.tabs.create(options);
+      };
 
-          // Re-focus the popup after 2 seconds to mitigate the problem of lost popups
-          // (e.g. when a user clicks the website)
-          setTimeout(() => {
-            if (!window.id) return;
+      const optionsPopup: browser.Windows.CreateCreateDataType = {
+        url: url,
+        type: "popup",
+        width: windowWidth,
+        height: windowHeight,
+        top: top,
+        left: left,
+      };
 
-            browser.windows.update(window.id, {
-              focused: true,
-            });
-          }, 2100);
-
-          const onMessageListener = (
-            responseMessage: {
-              response?: unknown;
-              error?: string;
-              data: Type;
-            },
-            sender: Runtime.MessageSender
-          ) => {
-            if (
-              responseMessage &&
-              responseMessage.response &&
-              sender.tab &&
-              sender.tab.id === tabId &&
-              sender.tab.windowId
-            ) {
-              browser.tabs.onRemoved.removeListener(onRemovedListener);
-              // if the window was opened as tab we remove the tab
-              // otherwise if a window was opened we have to remove the window.
-              // Opera fails to close the window with tabs.remove - it fails with: "Tabs cannot be edited right now (user may be dragging a tab)"
-              let closePromise;
-              if (closeWindow) {
-                closePromise = browser.windows.remove(sender.tab.windowId);
-              } else {
-                closePromise = browser.tabs.remove(sender.tab.id as number); // as number only for TS - we check for sender.tab.id in the if above
-              }
+      const optionsTab: browser.Tabs.CreateCreatePropertiesType = {
+        url: url,
+      };
 
-              return closePromise.then(() => {
-                // in the future actual "remove" (closing prompt) will be moved to component for i.e. budget flow
-                // https://github.com/getAlby/lightning-browser-extension/issues/1197
-                if (responseMessage.error) {
-                  return reject(new Error(responseMessage.error));
-                } else {
-                  return resolve(responseMessage);
-                }
-              });
-            }
-          };
+      const createPromise = browser.windows
+        ? createPopup(optionsPopup)
+        : createTab(optionsTab);
+
+      return createPromise.then(async (windowOrTab) => {
```

Comment:
Maybe we can use await instead of callbacks to help simplify this?

Reply:
we had callbacks before as well! i think we should do minimal possible changes here so that its easy to review the logic behind the code